### PR TITLE
rename searchFacet method

### PR DESCRIPTION
**Pull request by maxiloc**

### CODE REVIEW

**Comment by maxiloc** [comment]
@JanPetr @rayrutjes Can you guys review ?

File: src/AlgoliaSearch/Index.php [hunk 1]
```diff
@@ -1291,14 +1291,19 @@ public function saveSynonym($objectID, $content, $forwardToReplicas = false)
      */
     public function __call($name, $arguments)
     {
-        if ($name !== 'browse') {
-            return;
+        if ($name == 'browse') {
```
Comment (rayrutjes) [inline src/AlgoliaSearch/Index.php]: Let's do a strict check here for consistency

File: src/AlgoliaSearch/Index.php [hunk 2]
```diff
@@ -1291,14 +1291,19 @@ public function saveSynonym($objectID, $content, $forwardToReplicas = false)
...
+            if (count($arguments) >= 1 && is_string($arguments[0])) {
+                return call_user_func_array(array($this, 'doBrowse'), $arguments);
+            }
+
+            return call_user_func_array(array($this, 'doBcBrowse'), $arguments);
         }
 
-        if (count($arguments) >= 1 && is_string($arguments[0])) {
-            return call_user_func_array(array($this, 'doBrowse'), $arguments);
+        if ($name == 'searchFacet') {
```
Comment (rayrutjes) [inline src/AlgoliaSearch/Index.php]: Strict check as well

File: src/AlgoliaSearch/Index.php [hunk 3]
```diff
@@ -1291,14 +1291,19 @@ public function saveSynonym($objectID, $content, $forwardToReplicas = false)
...
+            if (count($arguments) >= 1 && is_string($arguments[0])) {
+                return call_user_func_array(array($this, 'doBrowse'), $arguments);
+            }
+
+            return call_user_func_array(array($this, 'doBcBrowse'), $arguments);
         }
 
-        if (count($arguments) >= 1 && is_string($arguments[0])) {
-            return call_user_func_array(array($this, 'doBrowse'), $arguments);
+        if ($name === 'searchFacet') {
```
Comment (rayrutjes) [inline src/AlgoliaSearch/Index.php]: I'm not a huge fan of this. We could simply keep the searchFacet() method and delegate the handling to searchForFacetValues().
that way we can easily flag as deprecated in the docblock and keeps the __call clean.

**Comment by rayrutjes** [comment]
LGTM

**Pull request merged**